src/openforms/submissions/logic/actions.py

```diff
@@ -308,6 +309,57 @@ def get_action_log_data(
         }
 
 
+class DMNConfig(TypedDict):
+    plugin_id: str
+    input_mapping: dict[str, str]
+    output_mapping: dict[str, str]
+    definition_id: str
+    definition_version: str
+
+
+@dataclass
+class EvaluateDMNAction(ActionOperation):
+    input_mapping: dict[str, str]
+    output_mapping: dict[str, str]
+    definition_id: str
+    plugin_id: str
+    definition_version: str = ""
+
+    @classmethod
+    def from_action(cls, action: ActionDict) -> "ActionOperation":
+        dmn_config: DMNConfig = action["action"]["value"]
+
+        return cls(**dmn_config)
+
+    def eval(
+        self,
+        context: DataMapping,
+        log: Callable[[JSONValue], None],
+        submission: Submission,
+    ) -> DataMapping | None:
+        # Mapping from form variables to DMN inputs
+        dmn_inputs = {
+            key_dmn: glom(context, key_form)
```

Comment:
this will probably cause problems with optional fields where deep nested keys don't exist? Any chance this looks like or can be brought into the shape of the `FormioData` data structure? It allows you to do `data[formio_key]` without having to worry about the `glom` implementation details.

Reply:
As I will probably need something similar for the Objects API ticket, wouldn't it make sense to have the `FormioData` user dict made "generic"? (By "generic" I just mean changing the name to `GlomableDict` or similar)

Reply:
careful - when I added that datastructure I tried updating the code to make use of it in more places and the test suite broke spectacularly :grimacing: it's probably a lot more work than you expect.

First I'd like to see this datastructure properly adapted everywhere and the raw `glom` usage replaced, then we can consider making it more generic. And about that... I'd like to hide the implementation details that `glom` is used and so not include `Glom` in the name. The purpose of the datastructure is mirroring how formio submission data is processed if there are <kbd>.</kbd> characters in their component/variable names, and glom is just a means to an end for that.

We're bound to the formio data model, and all the rest follows from that.

Reply:
> careful - when I added that datastructure I tried updating the code to make use of it in more places and the test suite broke spectacularly 😬 it's probably a lot more work than you expect.

yes we would have to be careful and look at the `lodash.get` implementation. In fact, `FormioData` is currently inconsistent with lodash as glom does not support `[0]` array indexing